### DISCUSSION TITLE
About downloading a bunch of works through bookmarks.

### DISCUSSION THREAD

**Discussion by RowanSkie** [nianeyna/ao3downloader#111] [Q&A] [answered]
So, I'm using this thing to download my bookmarked works and series. However, when it's off, it doesn't seem to download stories in bookmarked series. 

When it's on, however, it downloads *everything* on a series if the work is part of a series, leading to me at one point downloading 100 works for one series when I just bookmarked one piece for said series.

Should it do that? Or do I have to scan my bookmarks for the series and download them separately?

tl;dr: _When downloading the entire bookmarks list, should it also download bookmarked series without agreeing to "do you want to get works from series links? (y/n)"_

EDIT: If it doesn't, what if the program gives a .txt file of all bookmarked series then downloads from there?

**Comment by nianeyna** [comment]
Yeah, series link detection is kind of naive right now. If the downloader finds a link to a series anywhere on the page, it gets downloaded. It's not smart enough to tell whether it's a bookmark link or not. I really could have *sworn* I already had an open issue for this, because I've been meaning to work on it for ages. But uh apparently I never made a ticket for it. That has been addressed now lol. So, you can subscribe to #117 to get notified when this is fixed. For now you just kind of have to put up with it though I'm afraid. Maybe download without series expansion and then quickly page through and download any series bookmarks manually? (or via entering the individual series links into ao3downloader directly)

**Comment by nianeyna** [comment, marked as answer]
This issue is fixed in the latest version of the script - series bookmarks will now always be downloaded, regardless of whether you choose to detect series links or not
